client/blocks/reader-post-card/gallery.jsx

```diff
@@ -1,66 +1,111 @@
+import classnames from 'classnames';
 import PropTypes from 'prop-types';
+import { Component } from 'react';
 import ReaderExcerpt from 'calypso/blocks/reader-excerpt';
 import AutoDirection from 'calypso/components/auto-direction';
 import cssSafeUrl from 'calypso/lib/css-safe-url';
-import { isFeaturedImageInContent } from 'calypso/lib/post-normalizer/utils';
 import resizeImageUrl from 'calypso/lib/resize-image-url';
 import { imageIsBigEnoughForGallery } from 'calypso/state/reader/posts/normalization-rules';
 import { READER_CONTENT_WIDTH } from 'calypso/state/reader/posts/sizes';
 
-function getGalleryWorthyImages( post ) {
-	const numberOfImagesToDisplay = 4;
-	const images = ( post.images && [ ...post.images ] ) || [];
-	const indexToRemove = isFeaturedImageInContent( post );
-	if ( indexToRemove ) {
-		images.splice( indexToRemove, 1 );
-	}
+const noop = () => {};
 
-	return images.filter( imageIsBigEnoughForGallery ).slice( 0, numberOfImagesToDisplay );
-}
+class PostGallery extends Component {
+	state = {
+		itemSelected: 0,
+		listItems: null,
+	};
 
-const PostGallery = ( { post, children, isDiscover } ) => {
-	const imagesToDisplay = getGalleryWorthyImages( post );
-	const listItems = imagesToDisplay.map( ( image, index ) => {
-		const imageUrl = resizeImageUrl( image.src, {
-			w: READER_CONTENT_WIDTH / imagesToDisplay.length,
-		} );
-		const safeCssUrl = cssSafeUrl( imageUrl );
-		let imageStyle = { background: 'none' };
-		if ( safeCssUrl ) {
-			imageStyle = {
-				backgroundImage: 'url(' + safeCssUrl + ')',
-				backgroundSize: 'cover',
-				backgroundPosition: '50% 50%',
-				backgroundRepeat: 'no-repeat',
-			};
+	handleClick = ( event ) => {
+		event.preventDefault();
+		let updateItemSelected;
+		updateItemSelected = this.state.itemSelected + 1;
+		if ( updateItemSelected >= this.state.listItems.length ) {
+			updateItemSelected = 0;
 		}
+		this.setState( { itemSelected: updateItemSelected } );
+	};
+
+	getGalleryWorthyImages = ( post ) => {
+		const numberOfImagesToDisplay = 10;
+		const images = ( post.images && [ ...post.images ] ) || [];
+
+		return images
+			.sort( ( a, b ) => ( a.width > b.width ? 1 : -1 ) )
```

Comment:
The sort here was to actually filter out smaller images from the list of images in the post. I suspect that I can rewrite this to avoid sorting (it already sorts by how they are placed in the post afaik) and just filter out smaller/less suitable images that won't suit the gallery view.

Reply:
it looks like it still has the sort (maybe you didn't push 😆)